File: week1/CheckTheCheck/main.py
import sys

def main(*args):
    values = [int(x) for m in args[1::2] for x in m.split()]
    total = values.pop(-1)
    real_total = 0
    for i in range(0,len(values)-1, 2):
        real_total += values[i]*values[i+1]
    if real_total >= total:
        print("PAY")
    else:
        print("PROTEST")

    return 0

if __name__ == "__main__":
    # Read arguments from standard input (text file)
    lines = sys.stdin.readlines()

    main(*lines)

File: week1/CheckTheCheck/main.cpp
#include <cstdio>
#include <string>

int main() {
    long total_to_pay = 0;
    long total = 0;

    while (1){
        std::string dk;
        int ret = scanf("%[^\n]", dk);
        if (ret == 0) {
            break;
        }
        if (dk == "TOTAL") {
            scanf("%ld", &total_to_pay);
            break;
        }else{

            long price;
            scanf("%d", &price);
            total += price;
            printf("%s %d\n", dk , price);
        }

    }
    if (total_to_pay <= total) {
        printf("PAY\n");
    } else {
        printf("PROTEST\n");
    }

    return 0;
}